src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -422,31 +420,30 @@ internal float GetSingleWithQuotes()
                 return value;
             }
 
-            if (Utf8Parser.TryParse(span, out value, out int bytesConsumed)
-                && span.Length == bytesConsumed)
+            // NETCOREAPP implementation of the TryParse method above permits case-insensitive variants of the
+            // float constants "NaN", "Infinity", "-Infinity". This differs from the NETFRAMEWORK implementation.
+            // The following logic reconciles the two implementations to enforce consistent behavior.
+            if (!(Utf8Parser.TryParse(span, out value, out int bytesConsumed)
+                  && span.Length == bytesConsumed
+                  && JsonHelpers.IsFinite(value)))
             {
-                // NETCOREAPP implementation of the TryParse method above permits case-insenstive variants of the
-                // float constants "NaN", "Infinity", "-Infinity". This differs from the NETFRAMEWORK implementation.
-                // The following logic reconciles the two implementations to enforce consistent behavior.
-                if (JsonHelpers.IsFinite(value))
-                {
-                    return value;
-                }
+                ThrowHelper.ThrowFormatException(NumericType.Single);
+                Debug.Assert(false);
```

Comment:
Why do we need the Debug.Assert(false)?  There are _lots_ of places we simply expect ThrowHelper.ThrowXx to actually throw.